lms/djangoapps/verify_student/models.py

```diff
@@ -1112,8 +1112,13 @@ def get_or_create_verification_checkpoint(cls, course_id, checkpoint_location):
         Returns:
             VerificationCheckpoint object if exists otherwise None
         """
-        checkpoint, __ = cls.objects.get_or_create(course_id=course_id, checkpoint_location=checkpoint_location)
-        return checkpoint
+        try:
+            with transaction.atomic():
+                checkpoint, __ = cls.objects.get_or_create(course_id=course_id, checkpoint_location=checkpoint_location)
+                return checkpoint
+        except IntegrityError:
+            # Record already exists - just return it.
+            return cls.objects.get(course_id=course_id, checkpoint_location=checkpoint_location)
```

Comment:
@doctoryes This is always going to raise a `DidNotExist`. The `IntegrityError` happens because in a parallel request a row in the table has been created, however in the `get_or_create`:
1. The `get` will not see that row because it was committed after the transaction in which this `get` is being executed was started.
2. So `create` will be called. However, MySQL will raise an `IntegrityError` because it sees that a parallel request has already created the row and this transaction cannot be committed later.

So doing a `get` will fail with a `DoesNotExist.` In Django 1.4, we had a `commit()` call in the `except` clause which started a new transaction which could see the row and therefore the `get` here succeeded.